File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        length=float("inf")
        l=0
        r=0
        sum=0
        while r < len(nums):
            sum+=nums[r]
            if sum >= target:
                length=min(length,r-l+1)
            while sum >= target:
                sum-=nums[l]
                l+=1
                if sum>=target:
                    length=min(length,r-l+1)
            r+=1
        return 0 if length==float("inf") else length
        
           

        

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int result=Integer.MAX_VALUE;
        int l=0;
        int j;
        int curSum=0;
        for(j=0;j<nums.length;j++){
            curSum+=nums[j];
           // if(curSum >= target){
                
            while(l < nums.length && curSum>=target){
                result=Math.min(result,j-l+1);
            
                curSum-=nums[l];
                l++;
            }
            
        }
        if(result == Integer.MAX_VALUE) return 0;
        return result;
        
    }
}